checker/mta_sts.go

```diff
@@ -0,0 +1,149 @@
+package checker
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func filterByPrefix(records []string, prefix string) []string {
+	filtered := []string{}
+	for _, elem := range records {
+		if elem[0:len(prefix)] == prefix {
+			filtered = append(filtered, elem)
+		}
+	}
+	return filtered
+}
+
+func getKeyValuePairs(record string, lineDelimiter string,
+	pairDelimiter string) map[string]string {
+	parsed := make(map[string]string)
+	for _, line := range strings.Split(record, lineDelimiter) {
+		split := strings.Split(strings.TrimSpace(line), pairDelimiter)
+		if len(split) != 2 {
+			continue
+		}
+		key := strings.TrimSpace(split[0])
+		value := strings.TrimSpace(split[1])
+		if parsed[key] == "" {
+			parsed[key] = value
+		} else {
+			parsed[key] = parsed[key] + " " + value
+		}
+	}
+	return parsed
+}
+
+func checkMTASTSRecord(domain string) CheckResult {
+	result := CheckResult{Name: "mta-sts-txt"}
+	records, err := net.LookupTXT(fmt.Sprintf("_mta-sts.%s", domain))
```

Comment:
I've noticed rest of checker code use timeouts. This lookup seems to be not restricted.